source/uwp/Renderer/lib/DateTimeParser.cpp

```diff
@@ -29,9 +23,24 @@ std::string DateTimeParser::GenerateString(DateTimePreparser text)
         result.tm_mon = textSection->GetMonth();
         result.tm_year = textSection->GetYear() >= 1900 ? textSection->GetYear() - 1900 : 0;
 
+        DateTimePreparsedTokenFormat format = textSection->GetFormat();
+        if (format != DateTimePreparsedTokenFormat::RegularString && !haveSetLanguage)
+        {
+            std::locale language;
+            try
+            {
+                language = std::locale(m_languageString.c_str());
+            }
+            catch (...)
+            {
+                language = std::locale("");
+            }
```

Comment:
For every date we have to make the locale call, wouldnt it be better to have a flag in the DateTimePreparser that checks if at least one of the elements is a date that must be localized and so we could move this piece of code outside the for loop so the locale call will not be generated for each date?

Reply:
The haveSetLanguage boolean prevents us from calling this more than once.

---
In reply to: [182553935](https://github.com/Microsoft/AdaptiveCards/pull/1383#discussion_r182553935) [](ancestors = 182553935)

Reply:
That said, i do think your suggestion is cleaner, updated to use that approach

---
In reply to: [182561429](https://github.com/Microsoft/AdaptiveCards/pull/1383#discussion_r182561429) [](ancestors = 182561429,182553935)